docs/rfcs/0002-session-id-sharing.md

```diff
@@ -0,0 +1,226 @@
+- Status:
+- Date: 2023-06-07
+- Author: [@brizental](https://github.com/brizental)
+- RFC PR: [#7112](https://github.com/mozilla-mobile/mozilla-vpn-client/pull/7112)
+- Implementation GitHub issue: TODO
+
+## Problem Statement
+
+The VPN application is divided in two processes: the main application and the daemon. The main application is where the user interface runs and the daemon is what manages the VPN configuration and tunnel.
+
+In desktop environments, even while backgrounded the main application process is still running, but in mobile environments it is not. Due to this caveat, the mobile daemon processes have a separate instance of Glean to be able to collect telemetry about the connection while the application is in the background.
+
```

Comment:
Different topic: Sending telemetry "while the app is in the background" is, generally speaking, not ideal. Our use case here is different with the app rarely being opened/foregrounded once VPN is running, but we should work through how exactly (and as conservatively as possible) we're doing this. Do we perhaps already have a write-up of this aspect?

Reply:
There are three write ups related to this.
1. https://docs.google.com/document/d/1A2O_eACk0P3jDy0g2rRN5UTBOUj-iN-re6tVsRpxtDA/edit?usp=sharing
2. https://docs.google.com/document/d/1jyNZ_g_cUpZZsEr2hYwnwZgkxlvqTmoFUJKp_LAOPts/edit?usp=sharing
3. https://mozilla-hub.atlassian.net/wiki/spaces/SECPRV/pages/184419319/Metric+pings+-+Tech+spec

Would you mind expanding on "Sending telemetry "while the app is in the background" is, generally speaking, not ideal (on mobile)". I know about the challenges of spawning background tasks for the sake of submitting telemetry. However in this case, I believe that doesn't apply since we will be sending telemetry from an already running process.

Reply:
Got it, honestly performance is something I definitely overlooked when proposing and implementing the integration of Glean with the background processes. I have filed https://mozilla-hub.atlassian.net/browse/VPN-4965 so that we can deep dive on this.